### PR TITLE
Use SUPPORTED_CLIENT_FRAMEWORKS in templates

### PR DESCRIPTION
**Pull request by vivekmore**
-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/index.js [hunk 1]
```diff
@@ -82,10 +82,11 @@ module.exports = class extends BaseBlueprintGenerator {
                 }
             },
 
-            setupClientconsts() {
+            setupClientConstants() {
```
Comment (murdos) [inline generators/client/index.js]: This one is a breaking change. You should add it to https://github.com/jhipster/generator-jhipster/pull/10788 instead ;-)

File: generators/common/index.js [hunk 2]
```diff
@@ -49,13 +49,14 @@ module.exports = class extends BaseBlueprintGenerator {
                 this.checkInvocationFromCLI();
             },
 
-            setupConsts() {
+            setupConstants() {
```
Comment (murdos) [inline generators/common/index.js]: Breaking change too

File: generators/client/index.js (hunk 1 above)
Comment (vivekmore) [reply generators/client/index.js]: Ok I'll gladly revert and move these changes to the suggested PR.

These methods didn't appear to be used anywhere, hence I didn't think it'd be an issue.

Am I correct in understanding that you're saying it'll break blueprints and modules extending these core sub-generators?

Comment (murdos) [reply generators/client/index.js]: Thanks! Yes, you're correct: these methods could be used externally and thus should be considered as public API.

Comment (vivekmore) [reply generators/client/index.js]: Cool! 
I've moved the breaking changes to https://github.com/jhipster/generator-jhipster/pull/10788

**Pull request merged**